### PR TITLE
refactor: Extract data loading logic into class

### PR DESCRIPTION
**Pull request by TrevorBergeron**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

**Review by GarrettWu — COMMENTED** [review]
What a large PR. Do we have a bug number or any other contexts? What are we trying to solve here?

**Comment by TrevorBergeron** [comment]
> What a large PR. Do we have a bug number or any other contexts? What are we trying to solve here?

I do not have a bug number. Here is a previous commit in this effort: https://github.com/googleapis/python-bigquery-dataframes/commit/92fdb937220146f770c4ce32f8317eb55ece434d. 

The goal has been to make Session easier to understand by splitting it's state and functionality into distinct components. Mostly just moving code to new files right now. The hope is by creating some smaller components, it will be easier to optimize each component. Additionally, I do want to try making session thread-safe at some point, and pushing session state into individual objects will help with setting up necessary locks.

File: bigframes/session/loader.py [hunk 1]
```diff
@@ -0,0 +1,641 @@
...
+    def __init__(
+        self,
+        session: bigframes.session.Session,
+        bqclient: bigquery.Client,
+        storage_manager: bigframes.session.temp_storage.TemporaryGbqStorageManager,
+        default_index_type: bigframes.enums.DefaultIndexKind,
+        scan_index_uniqueness: bool,
+        metrics: Optional[bigframes.session.metrics.ExecutionMetrics] = None,
+    ):
+        self.bqclient = bqclient
```
Comment (GarrettWu) [inline bigframes/session/loader.py:103]: Some members are "private" while others are public. Are they really different or they can be consistent?

File: bigframes/session/loader.py [hunk 2]
```diff
@@ -0,0 +1,641 @@
...
+    This loader is constrained to loading local data and queries against data sources in the same region as the storage manager.
+    """
+
+    def __init__(
+        self,
+        session: bigframes.session.Session,
+        bqclient: bigquery.Client,
+        storage_manager: bigframes.session.temp_storage.TemporaryGbqStorageManager,
+        default_index_type: bigframes.enums.DefaultIndexKind,
+        scan_index_uniqueness: bool,
```
Comment (GarrettWu) [inline bigframes/session/loader.py:100]: Can we add docs for the args? At least scan_index_uniqueness and metrics aren't clear at first glance.

File: bigframes/session/executor.py [hunk 3]
```diff
@@ -147,6 +157,35 @@ def export(
             job_config=job_config,
         )
 
+    def export_gcs(
+        self,
+        array_value: bigframes.core.ArrayValue,
+        col_id_overrides: Mapping[str, str],
+        uri: str,
+        format: Literal["JSON", "CSV", "PARQUET"],
```
Comment (GarrettWu) [inline bigframes/session/executor.py:165]: nit: Python usually takes lower cases

File: bigframes/session/executor.py [hunk 4]
```diff
@@ -147,6 +157,35 @@ def export(
...
+        export_options: Dict[str, Union[bool, str]],
```
Comment (GarrettWu) [inline bigframes/session/executor.py:166]: Any chance to make these consistent? Mapping and Dict. Prefer Mapping as a more general type.

File: bigframes/session/executor.py (hunk 3 above)
Comment (TrevorBergeron) [reply bigframes/session/executor.py]: changed to lower case

File: bigframes/session/loader.py (hunk 2 above)
Comment (TrevorBergeron) [reply bigframes/session/loader.py:116]: added

File: bigframes/session/loader.py (hunk 1 above)
Comment (TrevorBergeron) [reply bigframes/session/loader.py]: made them all private

**Review by GarrettWu — APPROVED** [review]

**Pull request merged**